File: July 2022/July_05.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> hs;
        for(int i : nums)
            hs.insert(i);
        int ans = 0;
        for(int i : nums){
            if(hs.find(i - 1) == hs.end()){
                int curr = 1, j = i;
                while(hs.find(j + 1) != hs.end()){
                    j++;
                    curr++;
                }
                ans = max(ans, curr);
            }
        }
        return ans;
    }
};

File: July 2022/July_05.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int i : nums)
            set.add(i);
        int ans = 0;
        for(int i : nums){
            if(!set.contains(i - 1)){
                int curr = 0, j = i;
                while(set.contains(j)){
                    j++;
                    curr++;
                }
                ans = Math.max(ans, curr);
            }
        }
        return ans;
    }
}
